source/llassetgen/include/llassetgen/Packing.h

```diff
@@ -0,0 +1,176 @@
+#pragma once
+
+#include <algorithm>
+#include <cstdint>
+#include <iterator>
+#include <type_traits>
+#include <vector>
+
+#include <llassetgen/Geometry.h>
+#include <llassetgen/llassetgen_api.h>
+
+namespace llassetgen {
+    using PackingSizeType = unsigned int;
+
+    /*
+     * Describes the packing of a texture atlas.
+     *
+     * The rectangles in `rects` correspond to the input rectangles at the same
+     * position in the packing algorithms input.
+     */
+    struct Packing {
+        Vec2<PackingSizeType> atlasSize{};
+        std::vector<Rect<PackingSizeType>> rects{};
+
+        Packing() = default;
+
+        explicit Packing(Vec2<PackingSizeType> _atlasSize);
+    };
+
+    namespace impl_packing {
```

Comment:
Why underscore? For all variables you used camelCase. Please be consistent. Or is there any style rule that we agreed upon which I forgot? :D Is namespace naming an exception?

Reply:
The namespace was originally named `impl`, but I changed it to prevent potential name clashes should we use such a namespace in other headers. I now think that that is probably just overly cautious, and we should just stick with `impl` or `detail`. The most important part is that the name makes clear that the contents of the namespace are implementation details and not part of the libraries interface.

Reply:
`impl` and `detail` is also just the naming scheme I know from other C++ libraries, not sure if there is any best/most common choice here.